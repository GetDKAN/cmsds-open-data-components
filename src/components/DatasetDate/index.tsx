import React from 'react';
import DatasetDateItem from '../DatasetDateItem';

type DateObject = {
  modified?: string;
  released?: string;
  refresh?: string;
};

type DatasetDateProps = {
  /**
   * Released and modified date strings
   */
  date: DateObject;
  /**
   * Apply bold style to updated label
   */
  updatedBoldLabel?: boolean;
  /**
   * Apply bold style to released label
   */
  releasedBoldLabel?: boolean;
  /**
   * Apply bold style to refresh label
   */
  refreshBoldLabel?: boolean;
  /**
   * Display tooltips or not
   */
  displayTooltips?: boolean;
};

const DatasetDate: React.FC<DatasetDateProps> = (props) => {
  const {
    date,
    updatedBoldLabel = false,
    releasedBoldLabel = false,
    refreshBoldLabel = false,
    displayTooltips = true
  } = props;

  const { modified, released, refresh } = date;

  let count = 0;
  if (released) count++;
  if (modified) count++;
  if (refresh) count++;

  return (
    <div className='dataset-date'>
      {modified && (
        <DatasetDateItem 
          displayTooltips={displayTooltips} 
          type='modified' 
          date={modified} 
          boldLabel={updatedBoldLabel} 
        />
      )}
      {count > 1 && <span className='bullet-point'>&bull;</span>}
      {released && (
        <DatasetDateItem 
          displayTooltips={displayTooltips} 
          type='released' 
          date={released} 
          boldLabel={releasedBoldLabel} 
        />
      )}
      {count > 2 && <span className='bullet-point'>&bull;</span>}
      {refresh && (
        <DatasetDateItem 
          displayTooltips={displayTooltips} 
          type='refresh' 
          date={refresh} 
          boldLabel={refreshBoldLabel} 
        />
      )}
    </div>
  );
};

export default DatasetDate; 