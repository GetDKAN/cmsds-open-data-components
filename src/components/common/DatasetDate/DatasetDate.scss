@use "../../../scss/modules/modules";
@use "../../../scss/modules/colormap";

.dataset-date {
  display: flex;

  .dataset-date-item {
    color: colormap.$charcoal-90;
    font-size: 1.4rem;
    white-space: nowrap;

    &.bold-label > span {
      font-weight: 700;
    }
    .ds-c-tooltip__container {
      position: relative;
      bottom: 10px;
      width: 20px;
      display: inline-block;
    }
    .ds-c-tooltip__trigger-icon {
      padding: 0
    }
    .ds-c-tooltip {
      max-width: revert !important;
      font-size: var(--theme-font-size-md, 16px);
      font-style: normal;
      font-weight: 700;
      line-height: 150%; /* 24px */
    }
  }
}

@media screen and (max-width: 600px) {
  .dataset-date {
    flex-direction: column;

    .dataset-date-item {
      margin-bottom: 0.5rem;
    }

    .bullet-point {
      display: none;
    }
  }
}
