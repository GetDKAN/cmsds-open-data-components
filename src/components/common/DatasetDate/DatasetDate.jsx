import React from 'react'
import PropTypes from 'prop-types'
import DatasetDateItem from './DatasetDateItem/DatasetDateItem'

import './DatasetDate.scss'

const DatasetDate = (props) => {
  const {
    date,
    updatedBoldLabel = false,
    releasedBoldLabel = false,
    refreshBoldLabel = false,
    displayTooltips = true
  } = props

  const { modified, released, refresh } = date

  let count = 0
  released && count++
  modified && count++
  refresh && count++

  return (
    <div className='dataset-date'>
      {modified && <DatasetDateItem displayTooltips={displayTooltips} type='modified' date={modified} boldLabel={updatedBoldLabel} />}
      {count > 1 && <span className='bullet-point'>&bull;</span>}
      {released && <DatasetDateItem displayTooltips={displayTooltips} type='released' date={released} boldLabel={releasedBoldLabel} />}
      {count > 2 && <span className='bullet-point'>&bull;</span>}
      {refresh && <DatasetDateItem displayTooltips={displayTooltips} type='refresh' date={refresh} boldLabel={refreshBoldLabel} />}
    </div>
  )
}

DatasetDate.propTypes = {
  /**
   * Released and modified date strings
   */
  date: PropTypes.shape({
    modified: PropTypes.string,
    released: PropTypes.string,
    refresh: PropTypes.string
  }),
  /**
   * Apply bold style to updated label
   */
  updatedBoldLabel: PropTypes.bool,
  /**
   * Apply bold style to released label
   */
  releasedBoldLabel: PropTypes.bool,
  /**
   * Apply bold style to refresh label
   */
  refreshBoldLabel: PropTypes.bool,
  /**
   * Display tooltips or not
   */
  displayTooltips: PropTypes.bool
}
DatasetDate.displayName = 'DatasetDate'
export default DatasetDate
