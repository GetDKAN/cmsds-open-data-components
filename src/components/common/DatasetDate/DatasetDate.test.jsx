import renderer from 'react-test-renderer'
import DatasetDate from './DatasetDate'

describe('DatasetDate component.', () => {
  it('Matches snapshot.', () => {
    const renderedDatasetDate = renderer.create(
      <DatasetDate date={{
        modified: '2023-02-01',
        released: '2023-01-01'
      }} />
    ).toJSON()

    expect(renderedDatasetDate).toMatchSnapshot()
  })
})
