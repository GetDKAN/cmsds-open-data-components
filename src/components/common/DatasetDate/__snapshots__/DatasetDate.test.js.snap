// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetDate component. Matches snapshot. 1`] = `
<div
  className="dataset-date"
>
  <div
    className="dataset-date-item"
  >
    <span>
      Last Modified
    </span>
    : 
    February 1, 2023
  </div>
  <span
    className="bullet-point"
  >
    â€¢
  </span>
  <div
    className="dataset-date-item"
  >
    <span>
      Released
    </span>
    : 
    January 1, 2023
  </div>
</div>
`;
